pkg/api/validation/validation.go

```diff
@@ -1249,6 +1252,16 @@ func ValidateServiceAccountUpdate(oldServiceAccount, newServiceAccount *api.Serv
 	return allErrs
 }
 
+const SecretKeyFmt string = "\\.?" + util.DNS1123LabelFmt + "(\\." + util.DNS1123LabelFmt + ")*"
+
+var secretKeyRegexp = regexp.MustCompile("^" + SecretKeyFmt + "$")
+
+// IsSecretKey tests for a string that conforms to the definition of a
+// subdomain in DNS (RFC 1123), except that a leading dot is allowed
+func IsSecretKey(value string) bool {
```

Comment:
What are the secret keys used for? File names in secret volumes? Do they really need to have different validation rules than labels, annotations, and resources?

Note that we just tightened label rules.

cc @thockin